src/libstd/path.rs

```diff
@@ -0,0 +1,2415 @@
+    pub fn to_path_buf(&self) -> PathBuf {
+        PathBuf::from_path(self)
+    }
+
+    /// A path is *absolute* if it is indepedent of the current directory.
+    ///
+    /// * On Unix, a path is absolute if it starts with the root, so
+    /// `is_absolute` and `has_root` are equivalent.
+    ///
+    /// * On Windows, a path is absolute if it has a prefix and starts with the
+    /// root: `c:\windows` is absolute, while `c:temp` and `\temp` are not. In
+    /// other words, `path.is_absolute() == path.prefix().is_some() && path.has_root()`.
+    pub fn is_absolute(&self) -> bool {
+        self.has_root() &&
+            (cfg!(unix) || self.prefix().is_some())
+    }
+
+    /// A path is *relative* if it is not absolute.
+    pub fn is_relative(&self) -> bool {
+        !self.is_absolute()
+    }
+
+    /// Returns the *prefix* of a path, if any.
+    ///
+    /// Prefixes are relevant only for Windows paths, and consist of volumes
+    /// like `C:`, UNC prefixes like `\\server`, and others described in more
+    /// detail in `std::os::windows::PathExt`.
+    pub fn prefix(&self) -> Option<&Path> {
+        let iter = self.components();
+        optional_path(unsafe {
+            Path::from_u8_slice(
+                &self.as_u8_slice()[.. iter.prefix_remaining()])
+        })
+    }
+
+    /// A path has a root if the body of the path begins with the directory separator.
+    ///
+    /// * On Unix, a path hss a root if it begins with `/`.
+    ///
+    /// * On Windows, a path has a root if it:
+    ///     * has no prefix and begins with a separator, e.g. `\\windows`
+    ///     * has a prefix followed by a separator, e.g. `c:\windows` but not `c:windows`
+    ///     * has any non-disk prefix, e.g. `\\server\share`
+    pub fn has_root(&self) -> bool {
+         self.components().has_root()
+    }
+
+    /// The path without its final component.
+    ///
+    /// Does nothing, returnning `None` if the path consists of just a prefix
+    /// and/or root directory reference.
+    pub fn parent(&self) -> Option<&Path> {
+        let mut comps = self.components();
+        let comp = comps.next_back();
+        let rest = optional_path(comps.as_path());
+
+        match (comp, comps.next_back()) {
+            (Some(Component::CurDir), Some(Component::RootDir)) => None,
+            (Some(Component::CurDir), Some(Component::Prefix(_))) => None,
+            (Some(Component::Empty), Some(Component::RootDir)) => None,
+            (Some(Component::Empty), Some(Component::Prefix(_))) => None,
+            (Some(Component::Prefix(_)), None) => None,
+            (Some(Component::RootDir), Some(Component::Prefix(_))) => None,
+            _ => rest
+        }
+    }
+
+    /// The final component of the path, if it is a normal file.
+    ///
+    /// If the path terminates in `.`, `..`, or consists solely or a root of
+    /// prefix, `file` will return `None`.
+    pub fn file_name(&self) -> Option<&OsStr> {
```

Comment:
This is interesting returning `&OsStr` instead of `&Path`, I should probably read the RFC as to why, but is this mainly done to distinguish between "more paths" and "something that internally just looks like a path"?